### PR TITLE
Add dream and deps

### PR DESCRIPTION
**Pull request by ulrikstrid**
Do you want me to add overrides for all of caqti-*?

### CODE REVIEW

File: ocaml/dream/default.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      psq
+      result
+    ];
+
+    meta = {
+      description = " Easy-to-use, feature-complete Web framework without boilerplate";
+      license = lib.licenses.mit;
+    };
+  };
+}
```
Comment (aantron) [inline ocaml/dream/default.nix:65]: Ok sorry, I'm a third party here, but there is no newline at the end of this file! (Message not to be taken seriously).

File: ocaml/dream/default.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      yojson
+      # vendored dependencies, can we "unvendor" this?
+      gluten
+      gluten-lwt-unix
+      httpaf
+      httpaf-lwt-unix
+      h2
+      h2-lwt-unix
+      hpack
+      websocketaf
```
Comment (ulrikstrid) [inline ocaml/dream/default.nix:49]: This looks dangerous

Comment (anmonteiro) [reply ocaml/dream/default.nix:49]: I think these are not needed if they're vendored.

Comment (ulrikstrid) [reply ocaml/dream/default.nix]: Commented them out for now, I think @aantron wants to un-vendor them in  the long run.

File: ocaml/default.nix [hunk 3]
```diff
@@ -132,6 +132,16 @@ websocketafPackages // {
     doCheck = false;
   });
 
+  caqti = osuper.caqti.overrideAttrs (_: {
+    version = "1.5.1";
```
Comment (anmonteiro) [inline ocaml/default.nix:136]: unfortunately this doesn't really do anything, as version can't be overridden.

**Pull request merged**